### PR TITLE
Fix/misc minor bugs

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary

1.  Update requests exception for class instead of module
     When GitHub API wasn't responding, this output appeared about an exception:
     ```
       File "/opt/hostedtoolcache/Python/3.13.14/x64/lib/python3.13/site-packages/lasso/reports/corral/cattle_head.py", line 133, in _reachable                                              
       except requests.exceptions:                                                                                                                                                           
       logger.info(f"url {url} not reachable in {time_out}s")                                                                                                                                
       return False                                                                                                                                                                          
       TypeError: catching classes that do not inherit from BaseException is not allowed    
     ```
2. Replace eval() function
     According to Claude:
>      Source: Python documentation on https://docs.python.org/3/library/functions.html#getattr and https://docs.python.org/3/library/functions.html#eval which states: "This function can also be used to execute arbitrary code objects."    
>      
>      Why this is better:
>      1. Safe: getattr() is the standard Python way to access attributes/methods by name 
>      2. Clear intent: Shows we're doing attribute lookup, not arbitrary code execution
>      3. Better error handling: Raises AttributeError if method doesn't exist (vs obscure eval errors)                                                                                      

     
3. Fix logic to exclude dev builds when adding I&T reports
     Currently, it's injecting the I&T reports to files like `docs/source/releases/16.0-SNAPSHOT/index.rst`

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/lasso/reports/corral/cattle_head.py [hunk 1]
```diff
@@ -130,13 +130,14 @@ def _reachable(url):
         try:
             response = requests.head(url, timeout=time_out, headers=headers)
             return response.status_code != 404
-        except requests.exceptions:
+        except requests.exceptions.RequestException:
             logger.info(f"url {url} not reachable in {time_out}s")
```
Comment (nutjob4life) [inline src/lasso/reports/corral/cattle_head.py]: I know line 134 is not changed, but given the fix to line 133, it might be an opportunity to clean up line 134 as well and provide some more information:
```python
except requests.exceptions.RequestException as ex:
    logger.info("url %s not reachable in %ds: %s", url, time_out, ex)
```
This helps log the exception test, and also fixes the logging call which should always use `%` placeholders, no matter how nice and attractive those `f`-strings are!

File: src/lasso/reports/corral/cattle_head.py [hunk 2]
```diff
@@ -130,13 +130,14 @@ def _reachable(url):
...
             return False
 
     def _get_cell(self, function, format="md"):
         """Get the cell."""
-        link_func = eval(f"self._get_{function}_link()")
+        method_name = f"_get_{function}_link"
+        link_func = getattr(self, method_name)()
```
Comment (nutjob4life) [inline src/lasso/reports/corral/cattle_head.py:140]: Nice! This is much safer than using `eval`.

If you want to take it to the next level, you could have an "allow-list" of method names to check `method_name` against. But probably overkill.

**Review by nutjob4life — APPROVED** [review]
Code delta: ✓
Based on visual inspection: ✓ (trust accordingly)
Some comments: ✓ (interspersed)
Approval: ✅
However: note what `@Copilot` said, especially about `is_dev`

File: src/lasso/reports/corral/cattle_head.py (hunk 2 above)
Comment (c-suh) [reply src/lasso/reports/corral/cattle_head.py:140]: Have left it as-is since I'm going for expediency, but I appreciate the insight/suggestion and hope it'll stick in my brain!
